### PR TITLE
upgrade python and ubi base image

### PR DESCRIPTION
**Pull request by jasinner**
Upgrade python to 3.11 and ubi to 9.2

### CODE REVIEW

**Comment by jasinner** [comment]
I took inspiration from SDEngine for getting gssapi to compile.

**Comment by jasinner** [comment]
@RedHatProductSecurity/corgi-devs please review

**Review by JimFuller-RedHat — APPROVED** [review]
apart from CI failure - LGTM -  (works nicely locally for me) though lets wait for others to scan

File: pyproject.toml [hunk 1]
```diff
@@ -22,10 +22,6 @@ markers = [
     "performance",
     "unit",
 ]
-filterwarnings = [
-    # make warnings into errors that fail tests
-    "error",
-]
```
Comment (juspence) [inline pyproject.toml:28]: This shouldn't be removed - it's caught problems before that we didn't notice (e.g. incorrect use of files).

File: requirements/rpms.txt [hunk 2]
```diff
@@ -8,13 +8,10 @@ krb5-devel
...
 openssl
 openssl-devel
-python39
-python39-devel
-python39-pip
-python39-setuptools
-python39-wheel
+python3.11
+python3.11-devel
+python3.11-pip
```
Comment (juspence) [inline requirements/rpms.txt:16]: Are we missing -setuptools and -wheel, or are those no longer needed?

File: requirements/gssapi_build.in [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# See "pip install" section of Dockerfile for information about this requirements file.
+Cython >= 0.29.29, < 4.0.0
```
Comment (juspence) [inline requirements/gssapi_build.in:2]: Can we leave Cython unpinned in the .in file, or does that cause some problem?

File: requirements/gssapi_build.in [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+wheel
```
Comment (juspence) [inline requirements/gssapi_build.in:3]: If we keep python3.11-wheel in our list of RPM requirements, can we avoid including it separately in this GSSAPI-specific requirements file?

**Review by juspence — COMMENTED** [review]
LGTM other than a few changes.

**Pull request merged**